code/Path_Distance_TableGenerator.py

```diff
@@ -0,0 +1,311 @@
+    if len(results) > 1:
+        print("Encountered multiple city in the database", file=sys.stderr)
+        print(f"{src} matches multiple entries in the database. Please specify a state/province.", file=sys.stderr)
+        print("Matching cities:", file=sys.stderr)
+        for r in results:
+            print(f"\t{r[0]}, {r[1]}, {r[2]}", file=sys.stderr)
+        return False
+    if len(results) == 1:
+        fc = results[0][0]
+        fs = results[0][1]
+        fcc = results[0][2]
+
+    # verify the destination node is in the DB
+    results = global_querier.execute_query(t_query)
+    if len(results) == 0:
+        print(f"{dst} not found in the database.", file=sys.stderr)
+        return False
+    if len(results) > 1:
+        print("Encountered multiple city in the database", file=sys.stderr)
+        print(f"{dst} matches multiple entries in the database. Please specify a state/province.", file=sys.stderr)
+        print("Matching cities:", file=sys.stderr)
+        for r in results:
+            print(f"\t{r[0]}, {r[1]}, {r[2]}", file=sys.stderr)
+        return False
+    if len(results) == 1:
+        tc = results[0][0]
+        ts = results[0][1]
+        tcc = results[0][2]
+
+    # necessary if the user does not enter a state
+    src = f"{fc},{fs},{fcc}"
+    dst = f"{tc},{ts},{tcc}"
+
+    return True
+
+def query_db_for_nodes():
+    nodes_query = f"""SELECT * FROM city_points;"""
+    results = global_querier.execute_query(nodes_query)
+    for row in results:
+        city = row[0]
+        state = row[1]
+        country = row[2]
+        lat = float(row[3])
+        lon = float(row[4])
+        global_nodes_dict[(city, state, country)] = {}
+        global_nodes_dict[(city, state, country)]['x'] = lon
+        global_nodes_dict[(city, state, country)]['y'] = lat
+        global_nodes_dict[(city, state, country)]['GEOM'] = Point(lon, lat)
+
+def query_db_for_edges():
+    nodes_query = f"""SELECT * FROM standard_paths;"""
+    results = global_querier.execute_query(nodes_query)
+    for row in results:
+        fc = row[0]
+        fs = row[1]
+        fcc = row[2]
+        tc = row[3]
+        ts = row[4]
+        tcc = row[5]
+        dist_km = float(row[6])
+        path_wkt = row[7]
+        edge = ((fc, fs, fcc), (tc, ts, tcc))
+        global_edges_dict[edge] = {}
+        global_edges_dict[edge]['DIST_KM'] = dist_km
+        global_edges_dict[edge]['GEOM'] = wkt.loads(path_wkt)
+
+def create_graph():
+    # add nodes
+    for n in global_nodes_dict.keys():
+        x = global_nodes_dict[n]['x']
+        y = global_nodes_dict[n]['y']
+        geom = global_nodes_dict[n]['GEOM']
+        global_graph.add_node(n, x=x, y=y, geom=geom)
+    # add edges
+    for e in global_edges_dict.keys():
+        dist = global_edges_dict[e]['DIST_KM']
+        geom = global_edges_dict[e]['GEOM']
+        global_graph.add_edge(e[0], e[1], dist_km=dist, geom=geom)
+
+def initialize_the_global_graph(db_file):
```

Comment:
Can we put the graph related code into a class?

Reply:
actually most graph related function is using networkx inner function so I am afraid we don't have many of them to group as a class